lib/Client.js

```diff
@@ -149,6 +182,23 @@ class Client {
         this.client.editStatus("online", { name });
     }
 
+    clientReporting() {
+        this.database.insertStat("messageRate", this.state.messageCount);
+        this.database.insertStat("userCount", this.client.users.size);
+        this.database.insertStat("memoryUsage", process.memoryUsage().heapUsed);
+        this.database.insertStat("databaseLatency", Math.round(_.mean(this.database.latestLatencies)));
+        this.database.insertStat("shardLatency", this.getAverageShardLatency());
+        this.resetState();
+    }
+
+    getAverageShardLatency() {
+        const latency = [];
+        this.client.shards.forEach((shard) => {
+            latency.push(shard.latency);
+        });
+        return _.mean(latency);
```

Comment:
[Optional] this whole function could be `_.mean(this.client.shards.map(shard => shard.latency))` or some variation of it (like using a variable for the array of latency instead of all inline)

Reply:
Probably a good idea.